### PR TITLE
HDDS-10994. Migrate from Gradle Enterprise to Develocity

### PR DESCRIPTION
**Pull request by adoroszlai**
## What changes were proposed in this pull request?

Starting with version 1.21, the Gradle Enterprise extension is available under the “Develocity” brand.  This change [migrates](https://docs.gradle.com/develocity/maven-extension/legacy/#develocity_migration) configuration to allow future version bumps.

https://issues.apache.org/jira/browse/HDDS-10994

## How was this patch tested?

CI build in `apache/ozone`:
https://github.com/apache/ozone/actions/runs/9428877644/job/25974666271

Build scan published:
https://ge.apache.org/s/ez5vddmvqgzdm

CI run in fork:
https://github.com/adoroszlai/ozone/actions/runs/9429126848

### CODE REVIEW

**Comment by adoroszlai** [comment]
@clayburn please take a look if you have time

File: .mvn/develocity.xml [hunk 1]
```diff
@@ -19,22 +19,23 @@
...
-        xsi:schemaLocation="https://www.gradle.com/gradle-enterprise-maven https://www.gradle.com/schema/gradle-enterprise-maven.xsd">
+<develocity
+        xmlns="https://www.gradle.com/develocity-maven" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="https://www.gradle.com/develocity-maven https://www.gradle.com/schema/develocity-maven.xsd">
   <server>
     <url>https://ge.apache.org</url>
     <allowUntrusted>false</allowUntrusted>
   </server>
   <buildScan>
     <capture>
```
Comment (clayburn) [inline .mvn/develocity.xml:30]: This whole section can be omitted with the update. All of these values default to true. But you are welcome to leave them this way as well.

File: .mvn/develocity.xml [hunk 2]
```diff
@@ -19,22 +19,23 @@
...
 -->
-<gradleEnterprise
-        xmlns="https://www.gradle.com/gradle-enterprise-maven" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-        xsi:schemaLocation="https://www.gradle.com/gradle-enterprise-maven https://www.gradle.com/schema/gradle-enterprise-maven.xsd">
+<develocity
+        xmlns="https://www.gradle.com/develocity-maven" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="https://www.gradle.com/develocity-maven https://www.gradle.com/schema/develocity-maven.xsd">
   <server>
     <url>https://ge.apache.org</url>
     <allowUntrusted>false</allowUntrusted>
```
Comment (clayburn) [inline .mvn/develocity.xml:27]: Same for this value, the default is already false, so you can omit or leave the value.

**Review by clayburn — APPROVED** [review]
Looks good. I left a few stylistic commits on default values, but functionally this looks good.

**Comment by adoroszlai** [comment]
Thanks a lot @clayburn for the review.  For now, I prefer to keep these settings even if values match default ones.

**Review by fapifta — APPROVED** [review]
Thank you for working on this @adoroszlai, and thank you @clayburn for the review based on which I think we are good to go and commit this one.

**Comment by adoroszlai** [comment]
Thanks @clayburn, @fapifta for the review.

**Pull request merged**